src/main.c

```diff
@@ -442,6 +453,68 @@ main_loop(void)
 			free(phpcmd);
 		}
 
+		// set the protocol
+		if (config->fas_secure_enabled == 3) {
+			snprintf(protocol, sizeof(protocol), "https");
+		} else {
+			snprintf(protocol, sizeof(protocol), "http");
+		}
+
+		if (config->fas_remotefqdn) {
+			safe_asprintf(&fasurl, "%s://%s:%u%s",
+				protocol, config->fas_remotefqdn, config->fas_port, config->fas_path);
+			config->fas_url = safe_strdup(fasurl);
+		} else {
+			safe_asprintf(&fasurl, "%s://%s:%u%s",
+				protocol, config->fas_remoteip, config->fas_port, config->fas_path);
+			config->fas_url = safe_strdup(fasurl);
+		}
+		debug(LOG_NOTICE, "FAS URL is %s\n", config->fas_url);
+		free(fasurl);
+
+		if (config->fas_key && config->fas_secure_enabled == 3) {
+			// Check if authmon is already running and if it is, kill it
+			safe_asprintf(&fasssl, "kill $(pgrep -f authmon) > /dev/null 2>&1");
+			system(fasssl);
+			free(fasssl);
+
+			// Get the sha256 digest of gatewayname
+			safe_asprintf(&fasssl,
+				"echo \"<?php echo openssl_digest('%s', 'sha256'); ?>\" | %s",
+				config->gw_name,
+				config->fas_ssl
+				);
+
+			if (execute_ret_url_encoded(gwhash, sizeof(gwhash) - 1, fasssl) == 0) {
+				safe_asprintf(&gatewayhash, "%s", gwhash);
+				debug(LOG_INFO, "gatewayname digest is: %s\n", gwhash);
+			} else {
+				debug(LOG_ERR, "Error hashing gatewayname");
+				debug(LOG_ERR, "Exiting...");
+				exit(1);
+			}
+			free(fasssl);
+
+			// Start authmon in the background
+			safe_asprintf(&fasssl,
+				"/usr/lib/nodogsplash/authmon.sh \"%s\" \"%s\" \"%s\" &",
+				config->fas_url,
+				gatewayhash,
+				config->fas_ssl
+				);
```

Comment:
Please indent the closing parenthesis on the column of the function name. I think that is what do everywhere else.

Reply:
On checking, it seems it is not done everywhere else, but I agree it should be.